fdbrpc/TLSConnection.actor.cpp

```diff
@@ -299,43 +309,126 @@ void TLSOptions::register_network() {
 	new TLSNetworkConnections( Reference<TLSOptions>::addRef( this ) );
 }
 
+ACTOR static Future<ErrorOr<Standalone<StringRef>>> readEntireFile( std::string filename ) {
+	state Reference<IAsyncFile> file = wait(IAsyncFileSystem::filesystem()->open(filename, IAsyncFile::OPEN_READONLY, 0));
+	state int64_t filesize = wait(file->size());
+	state Arena arena(filesize+8);
```

Comment:
What's the +8 for?
Also this pattern is fine but there are some helpers for this type of thing so you can do:

state Standalone<StringRef> buf = makeString(fileSize);
int rc = wait(file->read(mutateString(buf), fileSize, 0));
..
return buf;

Reply:
I couldn't figure out any sizeof() magic to include the storage overhead of new[] potentially storing the number of elements requested.  I'll just do what you suggested though.